src/component/Card.jsx

```diff
@@ -10,12 +10,15 @@ function Card(props) {
     if (card1.value === card2.value) {
       console.log("Match Found");
     } else {
+      console.log("Not Match");
+
       setTimeout(() => {
         cardDeck[card1.id].flip = false;
         cardDeck[card2.id].flip = false;
-        console.log("Not Match");
         changeCardDeck(cardDeck);
-      }, 1000);
+        console.log(cardDeck);
+        forceUpdate();
```

Comment:
Instead of checking card Match here, Let it be on at top Like on Board component

Card component suppose to render cards & propagate flip event to it's parent

Reply:
ok, i will look into it. maybe after solving this re-render issue will solve.